src/main/resources/XWiki/SuggestDisplay.xml

```diff
@@ -258,13 +269,47 @@
     img.observe('click', this.removeItem.bindAsEventListener(this));
     li.appendChild(span);
     li.appendChild(hiddenInput);
+    //Add the edit button
+    var editImg = new Element("img",{'src': "$xwiki.getSkinFile('icons/silk/page_white_edit.png')"});
+    editImg.observe('click', this.editItem.bindAsEventListener(this));
+    //Check if in items editable mode
+    var inEditItemsMode = false;
+    var flag = Element.select($(this.list).up(0), ".inEditItemsMode");
```

Comment:
The code that checks for edit mode should be shared in one function to reduce risk and complexlty